src/parser/mod.rs

```diff
@@ -7506,7 +7506,7 @@ impl<'a> Parser<'a> {
                 with_offset,
                 with_offset_alias,
             })
-        } else if self.parse_keyword(Keyword::JSON_TABLE) {
+        } else if dialect_of!(self is MySqlDialect) && self.parse_keyword(Keyword::JSON_TABLE) {
```

Comment:
this is the actual change, right? It seems like the changes to derive/src/lib.rs are unrelated

Reply:
This sounds a little bit adhoc. Shouldn't this be a method on dialect ?

Reply:
JSON_TABLE seems to be implemented on oracle too, for instance: https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/JSON_TABLE.html#GUID-3C8E63B5-0B94-4E86-A2D3-3D4831B67C62

Reply:
And on DB2: https://www.ibm.com/docs/en/i/7.3?topic=data-using-json-table

Reply:
@alamb Yea, `derive/src/lib.rs` was changed by rustfmt. I restored it manually.

Reply:
We don't have dialects for Oracle and DB2.

Reply:
> This sounds a little bit adhoc. Shouldn't this be a method on dialect ?

This can be discussed separately, but this syntax follows existing ones.